### PR TITLE
Test time

### PR DESCRIPTION
**Pull request by maddieclayton**
## Description
https://github.com/Azure/azure-powershell/issues/5360

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Comment by maddieclayton** [comment]
Previous build (will have a few to test speed): https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/5472/

**Comment by vladimir-shcherbakov** [comment]
@praries880 Could you please take a look?

File: src/ResourceManager/Aks/Commands.Aks.Test/SessionRecords/Commands.Aks.Test.ScenarioTests.KubernetesTests/TestAzureKubernetes.json [hunk 1]
```diff
@@ -883,3619 +883,10 @@
...
-    {
-      "RequestUri": "/subscriptions/9e223dbe-3399-4e19-88eb-0975f02ac87f/providers/Microsoft.ContainerService/locations/eastus/operations/90be7340-30ee-4a03-b208-d91d6beaca92?api-version=2017-08-31",
-      "EncodedRequestUri": "L3N1YnNjcmlwdGlvbnMvOWUyMjNkYmUtMzM5OS00ZTE5LTg4ZWItMDk3NWYwMmFjODdmL3Byb3ZpZGVycy9NaWNyb3NvZnQuQ29udGFpbmVyU2VydmljZS9sb2NhdGlvbnMvZWFzdHVzL29wZXJhdGlvbnMvOTBiZTczNDAtMzBlZS00YTAzLWIyMDgtZDkxZDZiZWFjYTkyP2FwaS12ZXJzaW9uPTIwMTctMDgtMzE=",
-      "RequestMethod": "GET",
-      "RequestBody": "",
-      "RequestHeaders": {
-        "User-Agent": [
-          "FxVersion/4.7.2633.0",
-          "OSName/Windows10Enterprise",
-          "OSVersion/
```
Comment (praries880) [inline src/ResourceManager/Aks/Commands.Aks.Test/SessionRecords/Commands.Aks.Test.ScenarioTests.KubernetesTests/TestAzureKubernetes.json]: Is this all just the same operation being polled over n over again?

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Synchronize-AzureASInstance.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public class SynchronizeAzureAzureAnalysisServer : AzurePSCmdlet
     {
         private static TimeSpan DefaultPollingInterval = TimeSpan.FromSeconds(30);
```
Comment (praries880) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Synchronize-AzureASInstance.cs]: We don't need to set this to 0 in the tests?

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/ChangeLog.md [hunk 3]
```diff
@@ -20,7 +20,7 @@
 ## Current Release
 
 ## Version 0.6.8
-* Enable Gateway assocaite/disassociate operations on AS.
+* Enable Gateway assocaite/disassociate operations on AS.
```
Comment (praries880) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/ChangeLog.md]: revert this?

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/ComputeTestCommon.ps1 [hunk 4]
```diff
@@ -467,6 +467,18 @@ function Get-MarketplaceImage
     return $imgs;
 }
 
+function Get-FirstMarketPlaceImage
```
Comment (praries880) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/ComputeTestCommon.ps1]: where is this used?

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 5]
```diff
@@ -2244,13 +2236,10 @@ function Test-VirtualMachineWithPremiumStorageAccount
         # NRP
         $subnet = New-AzureRmVirtualNetworkSubnetConfig -Name ('subnet' + $rgname) -AddressPrefix "10.0.0.0/24";
         $vnet = New-AzureRmVirtualNetwork -Force -Name ('vnet' + $rgname) -ResourceGroupName $rgname -Location $loc -AddressPrefix "10.0.0.0/16" -Subnet $subnet;
-        $vnet = Get-AzureRmVirtualNetwork -Name ('vnet' + $rgname) -ResourceGroupName $rgname;
         $subnetId = $vnet.Subnets[0].Id;
         $pubip = New-AzureRmPublicIpAddress -Force -Name ('pubip' + $rgname) -ResourceGroupName $rgname -Location $loc -AllocationMethod Dynamic -DomainNameLabel ('pubip' + $rgname);
-        $pubip = Get-AzureRmPublicIpAddress -Name ('pubip' + $rgname) -ResourceGroupName $rgname;
         $pubipId = $pubip.Id;
         $nic = New-AzureRmNetworkInterface -Force -Name ('nic' + $rgname) -ResourceGroupName $rgname -Location $loc -SubnetId $subnetId -PublicIpAddressId $pubip.Id;
-        $nic = Get-AzureRmNetworkInterface -Name ('nic' + $rgname) -ResourceGroupName $rgname;
```
Comment (praries880) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.ps1]: how much time did we save when we removed these Get calls?

File: src/ResourceManager/ServiceBus/Commands.ServiceBus/ChangeLog.md [hunk 6]
```diff
@@ -21,7 +21,7 @@
 ## Current Release
 
 ## Version 0.6.5
-* Set minimum dependency of module to PowerShell 5.0
+* Set minimum dependency of module to PowerShell 5.0
```
Comment (praries880) [inline src/ResourceManager/ServiceBus/Commands.ServiceBus/ChangeLog.md]: revert this change?

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/SqlTestsBase.cs [hunk 7]
```diff
@@ -101,21 +101,13 @@ protected Management.Sql.SqlManagementClient GetSqlClient(RestTestFramework.Mock
             Management.Sql.SqlManagementClient client =
                 context.GetServiceClient<Management.Sql.SqlManagementClient>(
                     RestTestFramework.TestEnvironmentFactory.GetTestEnvironment());
-            if (HttpMockServer.Mode == HttpRecorderMode.Playback)
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/SqlTestsBase.cs]: why don't we need this anymore?

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Synchronize-AzureASInstance.cs (hunk 2 above)
Comment (maddieclayton) [reply src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Synchronize-AzureASInstance.cs]: Nope it isn't used in any tests.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/ChangeLog.md (hunk 3 above)
Comment (maddieclayton) [reply src/ResourceManager/AnalysisServices/Commands.AnalysisServices/ChangeLog.md]: Can't, it'll fail the checkchangelog script :(

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/ComputeTestCommon.ps1 (hunk 4 above)
Comment (maddieclayton) [reply src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/ComputeTestCommon.ps1]: In the test file below this. It removes like 99% of the file because it doesn't have recursively increasing piping.

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 5 above)
Comment (maddieclayton) [reply src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.ps1]: Not too much, but there's no real point to testing network within compute, and that's all the calls really do.

File: src/ResourceManager/ServiceBus/Commands.ServiceBus/ChangeLog.md (hunk 6 above)
Comment (maddieclayton) [reply src/ResourceManager/ServiceBus/Commands.ServiceBus/ChangeLog.md]: Can't because of changelog script.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/SqlTestsBase.cs (hunk 7 above)
Comment (maddieclayton) [reply src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/SqlTestsBase.cs]: We never needed it, it's set in the base class.  I think having it set redundantly here adds confusion for anyone looking at it.

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Comment by cormacpayne** [comment]
@maddieclayton can this PR be closed in favor of https://github.com/Azure/azure-powershell/pull/6559 ?

**Comment by maddieclayton** [comment]
Not right this second because I’m running tests to compare times to ensure we aren’t adding excessive time.

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Comment by maddieclayton** [comment]
@azuresdkci Test this please
